### ISSUE TITLE
Radio group in Forms doesn't respect properties

### ISSUE THREAD

**Issue by astor99** [mautic/mautic#13306]
### Mautic Version

5.0.x series

### Way of installing

I downloaded a release from https://www.mautic.org/mautic-releases

### PHP version

8.0.30

### What browsers are you seeing the problem on?

Not relevant

### What happened?

When I create a radio group in a form and select NO for "Use assigned contact/company field's list choices" with only one optionlist, I get the original fields anyway.
I suspect it happens for Checkbox group.
Sorry, my Mautic is in italian
![2024-02-02_11-31-44](https://github.com/mautic/mautic/assets/53852426/bbd3ecc2-91ae-4c12-8dcd-2a61d8dd31a5)

### How can we reproduce this issue?

Step 1: Create a form with a Radio Group
Step 2: Map a field which has got some options in "Mapped Field" tab, for example "Title" field.
Step 3:  Select NO in Properties tab on "Use assigned contact/company field's list choices" and add only one option list item
Step 4: Click on save or update
Step 5: You get Mr, Mrs, Miss choices anyway

### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

<br /><hr>
Care about this issue? Want to get it resolved sooner? If you are a <a href='https://www.mautic.org/become-a-member-of-mautic'>member of Mautic</a>, you can add some funds to the <a href='https://opencollective.com/mautic/projects/bounties'>Bounties Project</a> so that the person who completes this task can claim those funds once it is merged by a member of the core team! Read the docs <a href='https://contribute.mautic.org/product-team/mautic-bounty-programme'>here.</a>

**Comment by RCheesley** [comment]
Closing this in favour of the discussions on #13542 where it looks like there's some suggestions for resolving the issue - let's discuss it there.
